### PR TITLE
Add integration test for ingress addon

### PR DESCRIPTION
**Pull request by kairen**
The commit added test codes for Ingress addon, the feature request from #2174. This test will create a  ingress binding `nginx.example.com` domain name, and resolve to nginx service, finally use HTTP request to check response content.

For test result as below:
```sh
$ make integration
cp ./out/minikube-linux-amd64 ./out/minikube
go test -v -test.timeout=30m k8s.io/minikube/test/integration --tags="integration container_image_ostree_stub containers_image_openpgp" 
=== RUN   TestDocker
Environment=DOCKER_RAMDISK=yes FOO=BAR BAZ=BAT

ExecStart={ path=/usr/bin/dockerd ; argv[]=/usr/bin/dockerd -H tcp://0.0.0.0:2376 -H unix:///var/run/docker.sock --tlsverify --tlscacert /etc/docker/ca.pem --tlscert /etc/docker/server.pem --tlskey /etc/docker/server-key.pem --label provider=virtualbox --insecure-registry 10.96.0.0/12 --debug --icc=true ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }

--- PASS: TestDocker (73.77s)
=== RUN   TestFunctional
=== RUN   TestFunctional/Status
=== RUN   TestFunctional/DNS
=== RUN   TestFunctional/Logs
=== RUN   TestFunctional/Addons
=== RUN   TestFunctional/Dashboard
=== RUN   TestFunctional/IngressController
=== RUN   TestFunctional/ServicesList
=== RUN   TestFunctional/Provisioning
=== RUN   TestFunctional/EnvVars
=== RUN   TestFunctional/SSH
--- PASS: TestFunctional (16.45s)
    --- PASS: TestFunctional/Status (15.46s)
    	cluster_status_test.go:35: Checking if cluster is healthy.
    	cluster_status_test.go:45: Component: , Healthy: True.
    	cluster_status_test.go:45: Component: , Healthy: True.
    	cluster_status_test.go:45: Component: , Healthy: True.
    --- PASS: TestFunctional/Addons (0.24s)
    --- PASS: TestFunctional/SSH (0.55s)
    --- PASS: TestFunctional/EnvVars (0.81s)
    --- PASS: TestFunctional/ServicesList (1.05s)
    --- PASS: TestFunctional/Provisioning (1.39s)
    --- PASS: TestFunctional/Logs (2.15s)
    --- PASS: TestFunctional/Dashboard (9.78s)
    --- PASS: TestFunctional/DNS (12.18s)
    --- PASS: TestFunctional/IngressController (88.20s)
=== RUN   TestPersistence
--- PASS: TestPersistence (84.68s)
=== RUN   TestStartStop
--- PASS: TestStartStop (143.71s)
PASS
ok  	k8s.io/minikube/test/integration	406.872s
```

/cc @r2d4

### CODE REVIEW

**Comment by r2d4** [comment]
@minikube-bot ok to test

**Comment by kairen** [comment]
Hi @r2d4 I fixed the test fail issue, could you help me to retest this changes?

**Comment by kairen** [comment]
@r2d4 I have fixed the failed issue, now tests passed.

File: test/integration/addons_test.go [hunk 1]
```diff
@@ -68,6 +69,46 @@ func testDashboard(t *testing.T) {
 	}
 }
 
+func testIngressController(t *testing.T) {
+	t.Parallel()
+	minikubeRunner := NewMinikubeRunner(t)
+	kubectlRunner := util.NewKubectlRunner(t)
+
+	minikubeRunner.RunCommand("addons enable ingress", true)
```
Comment (r2d4) [inline test/integration/addons_test.go]: for future work, we should utilize the `minikube cache` command to make sure that the ingress controller image is cached for our tests.

Comment (kairen) [reply test/integration/addons_test.go]: @r2d4 Okay, I'll make this feature.

Comment (kairen) [reply test/integration/addons_test.go]: Added this feature from #2272.

**Pull request merged**